tap_github/scraping.py

```diff
@@ -0,0 +1,76 @@
+"""Utility functions for scraping https://github.com
+
+Inspired by https://github.com/dogsheep/github-to-sqlite/pull/70
+"""
+import requests
+import time
+
+
+def scrape_dependents(response: requests.Response, verbose=False):
+    from bs4 import BeautifulSoup
+
+    soup = BeautifulSoup(response.content, "html.parser")
+    # Navigate through Package toggle if present
+    options = soup.find_all("a", class_="select-menu-item")
+    links = []
+    if len(options) > 0:
+        for link in options:
+            links.append(link["href"])
+    else:
+        links.append(response.url)
+
+    if verbose:
+        print(links)
+
+    for link in links:
+        yield from _scrape_dependents(f"https://github.com/{link}", verbose=verbose)
+
+
+def _scrape_dependents(url, verbose=False):
```

Comment:
```suggestion
def _scrape_dependents(url: str, verbose: bool = False) -> Iterable[dict[str, Any]]:
```
so that the code can be checked by mypy. This probably needs some imports: `from typing import Any, Iterable`